packages/http/src/mocker/__tests__/HttpMocker.spec.ts

```diff
@@ -229,6 +228,118 @@ describe('HttpMocker', () => {
+          it('returns the first example', () => {
+            const response = httpMocker.mock({
+              input: mockInput,
+              resource: mockResource,
+              config: { mock: { dynamic: false } },
+            });
+
+            expect(response.body).toBeDefined();
+
+            const selectedExample = flatMap(mockResource.responses, res =>
+              flatMap(res.contents, content => content.examples || []),
+            )[0];
+
+            expect(selectedExample).toBeDefined();
+            expect(response.body).toEqual((selectedExample as INodeExample).value);
+          });
+        });
+
+        describe('and the resource has no response examples', () => {
+          const resourceOperation: IHttpOperation = {
+            id: 'id',
+            method: 'get',
+            path: '/test',
+            request: {},
+            responses: [
+              {
+                code: '200',
+                headers: [],
+                contents: [
+                  {
+                    mediaType: 'application/json',
+                    schema: {
+                      type: 'object',
+                      properties: {
+                        name: { type: 'string', examples: ['Clark'] },
+                        middlename: { type: 'string', examples: ['J'], default: 'JJ' },
+                        surname: { type: 'string', default: 'Kent' },
+                        age: { type: ['number', 'null'] },
+                        email: { type: 'string' },
+                        deposit: { type: 'number' },
+                        paymentStatus: { type: 'string', enum: ['completed', 'outstanding'] },
+                        creditScore: {
+                          anyOf: [{ type: 'number', examples: [1958] }, { type: 'string' }],
+                        },
+                        paymentScore: {
+                          oneOf: [{ type: 'string' }, { type: 'number', examples: [1958] }],
+                        },
+                        walletScore: {
+                          allOf: [{ type: 'string' }, { default: 'hello' }],
+                        },
+                        pet: {
+                          type: 'object',
+                          properties: {
+                            name: { type: 'string', examples: ['Clark'] },
+                            middlename: { type: 'string', examples: ['J'], default: 'JJ' },
+                          },
+                        },
+                      },
+                      required: ['name', 'surname', 'age', 'email'],
+                    },
+                    encodings: [],
+                  },
+                ],
+              },
+            ],
+          };
+
+          const response = httpMocker.mock({
+            input: mockInput,
+            resource: resourceOperation,
+            config: { mock: { dynamic: false } },
+          });
+
+          describe('the property has an example key', () => {
```

Comment:
By the time of reading this line scrolling up and down to see what the given object was is becoming painful. I can feel my mouse scroll temperature increased!

Reply:
As in you had to go up and forth too many times to check out the property you were testing?

Reply:
Makes sense, I can probably refactor it

Reply:
7123262ed82371f969c4af2a8d98e4ccfa3af588